sql/hive/src/main/scala/org/apache/spark/sql/hive/MetastoreRelation.scala

```diff
@@ -140,7 +145,12 @@ private[hive] case class MetastoreRelation(
           sparkSession.sessionState.conf.defaultSizeInBytes
         })
     }
-  )
+    if (catalogTable.catalogStats.isDefined) {
```

Comment:
Actually the `catalogStats` here is already obtained from Hive's number in `constructStatsFromHive` below.